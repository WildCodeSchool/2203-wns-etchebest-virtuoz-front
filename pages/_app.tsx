import "../styles/globals.css";
import fetch from "cross-fetch";
import type { AppProps } from "next/app";
import dynamic from "next/dynamic";
import {
  ApolloClient,
  InMemoryCache,
  ApolloProvider,
  gql,
  HttpLink,
} from "@apollo/client";

const client = new ApolloClient({
  link: new HttpLink({ uri: "http://localhost:4000/", fetch }),
  cache: new InMemoryCache(),
});

export const users = gql`
  query getAllUsers {
    getAllUsers {
      email
      name
    }
  }
`;

export const status = gql`
  query getAllStatus {
    getAllStatus {
      name
    }
  }
`;

function MyApp({ Component, pageProps }: AppProps) {
  return (
    <ApolloProvider client={client}>
      <Component {...pageProps} />
    </ApolloProvider>
  );
}

// Disabling SSR
export default dynamic(() => Promise.resolve(MyApp), { ssr: false });
