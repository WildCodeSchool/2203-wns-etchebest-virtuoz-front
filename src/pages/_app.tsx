import '../styles/globals.css';
import fetch from 'cross-fetch';
import type { AppProps } from 'next/app';
import dynamic from 'next/dynamic';
import {
  ApolloClient,
  InMemoryCache,
  ApolloProvider,
  gql,
  HttpLink,
  createHttpLink,
} from '@apollo/client';
import { setContext } from '@apollo/client/link/context';

// const client = new ApolloClient({
//   link: new HttpLink({ uri: "http://localhost:4000/", fetch }),
//   cache: new InMemoryCache(),
//   link: authLink.concat(httpLink),
// });
const httpLink = createHttpLink({
  uri: 'http://localhost:4000/graphql',
});

const authLink = setContext((_, { headers }) => {
  const token = localStorage.getItem('token');
  return {
    headers: {
      ...headers,
      authorization: token,
    },
  };
});

const client = new ApolloClient({
  cache: new InMemoryCache(),
  link: authLink.concat(httpLink),
});

export const users = gql`
  query GetAlluser {
    getAllUsers {
      email
      name
    }
  }
`;

export const createUser = gql`
  mutation Mutation($name: String, $email: String, $password: String) {
    createUser(name: $name, email: $email, password: $password) {
      email
      name
      password
    }
  }
`;

export const status = gql`
  query getAllStatus {
    getAllStatus {
      name
    }
  }
`;

function MyApp({ Component, pageProps }: AppProps) {
  return (
    <ApolloProvider client={client}>
      <Component {...pageProps} />
    </ApolloProvider>
  );
}

// Disabling SSR
export default dynamic(() => Promise.resolve(MyApp), { ssr: false });
