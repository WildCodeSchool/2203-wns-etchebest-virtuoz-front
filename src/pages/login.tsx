import { useState } from 'react';
import { useLazyQuery, gql } from '@apollo/client';
import { useRouter } from 'next/router';
import styles from '../styles/login.module.css';

const LOGIN = gql`
  query login($email: String!, $password: String!) {
    login(email: $email, password: $password)
  }
`;

const Login = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const [getToken, { data }] = useLazyQuery(LOGIN);
  const router = useRouter();
  console.log('data :', data);
  if (data) {
    console.log(data.login);
    localStorage.setItem('token', data.login);
  }
  return (
    <div className={styles.container}>
      <h3>Login</h3>
      <input className={styles.input}
        placeholder='email'
        value={email}
        onChange={(e) => setEmail(e.target.value)}
      />
      <input className={styles.input2}
        placeholder='password'
        value={password}
        onChange={(e) => setPassword(e.target.value)}
      />
      <button
        onClick={async () => {
          try {
            await getToken({ variables: { email: email, password: password } });
            router.push('/');
          } catch (err) {
            console.log('Handle me', err);
          }
        }}
      >
        Login
      </button>
    </div>
  );
};

export default Login;
