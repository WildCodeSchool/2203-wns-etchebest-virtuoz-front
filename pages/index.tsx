import { useQuery } from "@apollo/client";
import type { NextPage } from "next";
import Head from "next/head";
import styles from "../styles/Index.module.css";
import { users, status } from "./_app";

const Home: NextPage = () => {
  const getAllStatus = useQuery(status);
  const getAllUsers = useQuery(users);

  if (getAllStatus.loading || getAllUsers.loading) return <p>Loading...</p>;
  if (getAllStatus.error || getAllUsers.error) return <p>Error :(</p>;

  function testNum({ a }: { a: number }): string {
    let result;
    const newLocal = (result = a > 0 ? "positive" : "NOT positive");
    return newLocal;
  }

  console.log(testNum({ a: -5 }));
  return (
    <div className={styles.container}>
      <Head>
        <title>Create Next App</title>
        <meta name="description" content="Generated by create next app" />
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <main className={styles.main}>
        {getAllStatus.data.allStatus.map(({ name }: { name: any }) => (
          <div key={name}>
            <p>{name}</p>
          </div>
        ))}
        {getAllUsers.data.getAllUsers.map(
          ({ email, name }: { email: any; name: any }) => (
            <div key={name}>
              <p>
                {email}: {name}
              </p>
            </div>
          )
        )}
      </main>
    </div>
  );
};

export default Home;
