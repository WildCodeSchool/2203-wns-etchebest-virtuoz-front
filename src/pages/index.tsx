import { useQuery } from "@apollo/client";
import type { NextPage } from "next";
import Head from "next/head";
import TasksBoardViews from "../views/TasksBoardViews";
import styles from "../styles/Index.module.css";
import { users } from "./_app";

const Home: NextPage = () => {
  // const getAllStatus = useQuery(status);
  const getAllUsers = useQuery(users);

  if (getAllUsers.loading) return <p>Loading...</p>;
  if (getAllUsers.error) return <p>Error :::(</p>;

  function testNum({ a }: { a: number }): string {
    let result;
    const newLocal = (result = a > 0 ? "positive" : "NOT positive");
    return newLocal;
  }

  console.log(testNum({ a: -5 }));
  return (
    <div className={styles.container}>
      <Head>
        <title>Create Next App</title>
        <meta name="description" content="Generated by create next app" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <p>TEST test</p>
      <main className={styles.main}>
        <TasksBoardViews />
        {/* {getAllUsers.data.getAllUsers.map(
          ({ email, name }: { email: any; name: any }) => (
            <div key={name}>
              <p>
                {email}: {name}
              </p>
            </div>
          )
        )} */}
      </main>
    </div>
  );
};

export default Home;
