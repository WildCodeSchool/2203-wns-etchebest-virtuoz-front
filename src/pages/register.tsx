import { gql, useLazyQuery, useMutation } from "@apollo/client";
import { Router } from "next/router";
import React, { useState } from "react";
import { createUser } from "./_app";
import { useRouter } from "next/router";

const REGISTER = gql`
  query Query($name: String, $email: String, $password: String) {
    createUser(name: $name, email: $email, password: $password)
  }
`;
const Registration = () => {
  const router = useRouter();
  const [name, setName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const [register, { data }] = useLazyQuery(REGISTER);

  if (data) {
    console.log(data.createUser);
    localStorage.setItem("token", data.createUser);
  }

  return (
    <div className="registrationPage">
      <div className="form">
        <form
          onSubmit={async (e) => {
            try {
              e.preventDefault();
              // addUser({
              //   variables: { name: name, email: email, password: password },
              // });
              await register({
                variables: { email: email, password: password, name: name },
              });
              setName("");
              setEmail("");
              setPassword("");
              alert("Registered");
              router.push("/");
            } catch (err) {
              console.log("Handle me", err);
            }
          }}
        >
          <p className="title">Formulaire d&apos;inscription</p>
          <div className="form-inner">
            <div>
              <input
                type="text"
                name="username"
                id="firstname"
                placeholder="Prénom"
                onChange={(e) => setName(e.target.value)}
                value={name}
                className="RegistrationInput"
              />
            </div>
            <div>
              <input
                type="text"
                name="email"
                id="email"
                placeholder="E-mail"
                onChange={(e) => setEmail(e.target.value)}
                value={email}
                className="RegistrationInput"
              />
            </div>
            <div>
              <input
                type="password"
                name="Password"
                id="Password"
                placeholder="Mot de passe"
                onChange={(e) => setPassword(e.target.value)}
                value={password}
                className="RegistrationInput"
              />
            </div>
            <div>
              <button
                type="submit"
                value="Continuer"
                className="buttonRegistration"
              >
                Register{" "}
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  );
};

export default Registration;
