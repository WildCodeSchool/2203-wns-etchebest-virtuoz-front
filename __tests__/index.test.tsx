import { findByText, render, screen, waitFor } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import "@testing-library/jest-dom";
import { MockedProvider } from "@apollo/client/testing";
import { users } from "pages/_app";
import Home from "pages";

const mocks = [
  {
    request: {
      query: users,
    },
    result: {
      data: {
        getAllUsers: [
          {
            id: "1",
            email: "test",
            name: "sdi",
            password: "test",
          },
          {
            id: "2",
            email: "@moss",
            name: "Moss",
            password: "mosspassword",
          },
        ],
      },
    },
  },
];

describe("Home", () => {
  it("renders a heading", async () => {
    render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Home />
      </MockedProvider>
    );
    const element = await waitFor(() => screen.getByText("@moss: Moss"));
    expect(element).toBeInTheDocument();
  });
});
