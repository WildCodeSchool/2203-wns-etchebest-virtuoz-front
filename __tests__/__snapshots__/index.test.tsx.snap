// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a heading 1`] = `
<p>
  Loading...
</p>
`;
